Makefile

```diff
@@ -64,27 +67,88 @@ build-env:
+	${SYMLINK_FILE} -t nx-X11/exports/include/nx-X11/Xtrans/ ../../../../lib/include/xtrans/Xtrans.h
+	${SYMLINK_FILE} -t nx-X11/exports/include/nx-X11/Xtrans/ ../../../../lib/include/xtrans/Xtransint.h
+	${SYMLINK_FILE} -t nx-X11/exports/include/nx-X11/Xtrans/ ../../../../lib/include/xtrans/Xtranslcl.c
+	${SYMLINK_FILE} -t nx-X11/exports/include/nx-X11/Xtrans/ ../../../../lib/include/xtrans/Xtranssock.c
+	${SYMLINK_FILE} -t nx-X11/exports/include/nx-X11/Xtrans/ ../../../../lib/include/xtrans/Xtransutil.c
+
+	cd nx-X11 && make BuildDepends FONT_DEFINES=$(FONT_DEFINES)
+
+clean-env: version
+	${RM_FILE} nx-X11/exports/include/nx-X11/Xlib.h			\
+	      nx-X11/exports/include/nx-X11/Xresource.h			\
+	      nx-X11/exports/include/nx-X11/Xutil.h			\
+	      nx-X11/exports/include/nx-X11/cursorfont.h		\
+	      nx-X11/exports/include/nx-X11/Xlibint.h			\
+	      nx-X11/exports/include/nx-X11/Xcms.h			\
+	      nx-X11/exports/include/nx-X11/Xlocale.h			\
+	      nx-X11/exports/include/nx-X11/XKBlib.h			\
+	      nx-X11/exports/include/nx-X11/XlibConf.h			\
+	      nx-X11/exports/include/nx-X11/Xregion.h			\
+	      nx-X11/exports/include/nx-X11/ImUtil.h			\
+	      $(NULL)
+	${RM_FILE} nx-X11/exports/include/nx-X11/Xtrans/transport.c	\
+	      nx-X11/exports/include/nx-X11/Xtrans/Xtrans.c		\
+	      nx-X11/exports/include/nx-X11/Xtrans/Xtrans.h		\
+	      nx-X11/exports/include/nx-X11/Xtrans/Xtransint.h		\
+	      nx-X11/exports/include/nx-X11/Xtrans/Xtranslcl.c		\
+	      nx-X11/exports/include/nx-X11/Xtrans/Xtranssock.c		\
+	      nx-X11/exports/include/nx-X11/Xtrans/Xtransutil.c		\
+	      $(NULL)
+	-rmdir exports/include/nx-X11/Xtrans/
+	-rmdir exports/include/nx-X11/
+
 	# prepare Makefiles and the nx-X11 symlinking magic
-	cd nx-X11 && make BuildEnv FONT_DEFINES=$(FONT_DEFINES)
+	cd nx-X11 && make CleanEnv FONT_DEFINES=$(FONT_DEFINES)
 
 build-lite:
 	cd nxcomp && autoreconf -vfsi && (${CONFIGURE}) && ${MAKE}
 	cd nxproxy && autoreconf -vfsi && (${CONFIGURE}) && ${MAKE}
 
 build-full: build-env
 # in the full case, we rely on "magic" in the nx-X11 imake-based makefiles...
+
+	# build nxcomp first
 	cd nxcomp && autoreconf -vfsi && (${CONFIGURE}) && ${MAKE}
-	# build libNX_X11 and libNX_Xext prior to building
-	# nxcomp{ext,shad}.
-	cd nx-X11/lib && make
 
-	cd nxcompshad && autoreconf -vfsi && (${CONFIGURE}) && ${MAKE}
+	# build libNX_X11 second
+	cd nx-X11/lib && autoreconf -vfsi && (${CONFIGURE} --disable-polling) && ${MAKE} && cd - || exit -1
+	mkdir -p nx-X11/exports/lib/
+	$(SYMLINK_FILE) -t nx-X11/exports/lib/ ../../lib/src/.libs/libNX_X11.so
+	$(SYMLINK_FILE) -t nx-X11/exports/lib/ ../../lib/src/.libs/libNX_X11.so.6
+	$(SYMLINK_FILE) -t nx-X11/exports/lib/ ../../lib/src/.libs/libNX_X11.so.6.3.0
 
-	./mesa-quilt push -a
+	# build nxcompshad third
+	cd nxcompshad && autoreconf -vfsi && (${CONFIGURE}) && ${MAKE} && cd - || exit -1
 
+	# build nxagent fourth
+	./mesa-quilt push -a
```

Comment:
> [ -x ./mesa-quilt ] is missing here

No, it is not. In the clean target, we check for presence of mesa-quilt, because the clean target is the same for nx-libs and nx-libs-lite. And in case of nx-libs-lite, mesa-quilt is not presence.

Whereas nx-libs (the non-lite tarball) always has mesa-quilt. It would fail without. So in theory, no need to check. In fact, a missing mesa-quilt would be a good reason for the nx-libs build to fail.